src/types.h

```diff
@@ -567,13 +567,70 @@ namespace librealsense
     ////////////////////////////////////////////
     // World's tiniest linear algebra library //
     ////////////////////////////////////////////
-#pragma pack(push, 1)
-    struct int2 { int x, y; };
-    struct float2 { float x, y; float & operator [] (int i) { return (&x)[i]; } };
-    struct float3 { float x, y, z; float & operator [] (int i) { return (&x)[i]; } };
-    struct float4 { float x, y, z, w; float & operator [] (int i) { return (&x)[i]; } };
-    struct float3x3 { float3 x, y, z; float & operator () (int i, int j) { return (&x)[j][i]; } }; // column-major
-    struct pose { float3x3 orientation; float3 position; };
+#pragma pack( push, 1 )
+    struct int2
+    {
+        int x, y;
+    };
+    struct float2
+    {
+        float x, y;
+        float & operator[]( int i )
+        {
+#ifdef _DEBUG
+
+            assert( i >= 0 );
+            assert( i < 2 );
+#endif
```

Comment:
Will not affect release builds and therefore the undefined behavior cannot be mitigated.
Couples of ways to consider :
- Compile time checks with scoped enumeration types (lots of code to refactor with unpleasant casts)
- Run-time checks by switching to `std::array::at(i)` implementation

Reply:
@ev-mp  I didn't want to increase runtime and to catch issues on development.
@maloel what do you think?

Reply:
I think anything is better than what was there before.
Runtime shouldn't be affected (in release), it's too low-level.
Nir, please make sure asserts are removed in RelWithDebInfo. Check if CppCheck runs with _DEBUG...
If asserts are removed without _DEBUG (and _DEBUG is not in RelWithDebInfo), I see no issue with this.

Reply:
Checked.
- _DEBUG is not set on RelWithDebInfo
- Test fix with CppCheck 2.4.1 - no errors
- Manual tests to verify the behavior have not been change (removed from the PR)

Update pushed